### ISSUE TITLE
Parts that are hidden should not render DOM nodes

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#169529]
While talking with @deepak1556 he made me aware that the title bar part would still render DOM nodes such as the `window-control-container` even when native title is enabled.

It seems, a hidden or not-enabled part still adds DOM nodes, which seems odd. I tried also with activity bar and noticed that DOM nodes are there:

![image](https://user-images.githubusercontent.com/900690/208421091-46bdb889-3637-41bb-967b-679b5fbd4553.png)

I would strongly suggest that a hidden or disabled workbench part does not add DOM nodes and is fully excluded from the workbench layout.

**Comment by sbatten** [comment]
I could be misremembering, but I feel like for parts that toggle visibility, we got better performance when toggling them by using this approach. @joaomoreno I believe was part of that.

**Comment by bpasero** [comment]
Yeah good point, for components that typically toggle often such as sidebar, it should be optimized for speed. But parts that are likely remaining disabled, I would remove the DOM nodes.

Btw this came from looking into the slow window resize performance: the CSS rule that was slow for the window control container was still applied because the `div` was in the DOM even when native title was enabled.

**Comment by sbatten** [comment]
@bpasero does that mean even if the rule exists but doesn't apply to an existing element, it will be slow?

**Comment by bpasero** [comment]
I did not confirm that to be honest, but people have indicated that perf was slow even with native title, so it was a suspicion.

**Comment by deepak1556** [comment]
@sbatten I was only able to confirm for the `env(titlebar-area-*)` case where the presence of `window-controls-container` even in native titlebar scenario cause the same execessive style recalc, so it seems to have effect. I cannot talk for the other cases tho, it needs to be measured :)

**Comment by sbatten** [comment]
I think the biggest problem here is that the titlebar part is rendering in the native case, where it will never show. Other parts that can be toggled but are less likely to be can be optimized as well, but I'll focus on the native titlebar case where a lot of logic and dom rendering are still coming into play. Due to the layout code, I will probably implement a NativeTitlebarPart to provide the ITitleService interface and a really cheap serializable view.
